### ISSUE TITLE
.wav "import failed"

### ISSUE THREAD

**Issue by celiandu61** [LibreTime/libretime#279]
Hey, since some days i'm trying to import lot of files but a i have a problem with .wav i got everytime "import failed". It's not a limit of size because i can upload big mp3 file ( 114Mo ) without any problem but i can't upload a 32Mo .wav. 

If tried to convert a .wav to .flac using : 
`ffmpeg -i input.wav -af aformat=s32:44100 out.flac` 
And the upload worked so it's really a problem with .wav

If someone have an idea don't hesitate.

**Comment by hairmare** [comment]
Sorry for the late reply, I was rather busy the last couple of weeks.

.wav support should be supported, but I haven't tested it with a large amount of files.

Is analyzer logging anything? If it recognizes the file as WAV and has issues it should log "Invalid WAVE file.".

**Comment by celiandu61** [comment]
Sorry, my server was down since a week. I just retry and i found that in the analyzer : `2017-09-03 07:32:13,780 [metadata_analyzer] [ERROR]  Invalid WAVE file.
`
But this file is perfectly readable on VLC.

**Comment by hairmare** [comment]
It could be failing due to any [of these lines](https://github.com/LibreTime/libretime/blob/58875a2187db0f6dfdc497f2177a16841ca2d77d/python_apps/airtime_analyzer/airtime_analyzer/metadata_analyzer.py#L169-L177) in analyzer. Most likely the file is missing some metadata and the code is way to strict about it. 

We should probably improve the error message by adding some info on what exactly failed and maybe also make the code a bit more tolerant to missing metadata.

Can you share the wav to help in debugging or is it copyrighted?

**Comment by celiandu61** [comment]
Sorry again for the time to reply. I found a file with the problem ( i have the permission of the author ) :) 

https://www.dropbox.com/s/42a2nwd5u7kykuf/ROB%27S%20-%20Candy%20Floss%20%28WIP%29.wav?dl=0

It's probably a stupid question but i don't understand exactly the try part, where/when the  "except wave.Error:" is define in case of fail ?

**Comment by hairmare** [comment]
Thanks for the example, I can reproduce your issue with it.

According to [this stackoverflow](https://stackoverflow.com/a/27101840) the python wave module only supports PCM wav files.

When I tried reading the file with the wave module in an isolated test case I got the following error:

```
wave.Error: unknown format: 3
```

According to [this document](http://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/WAVE.html) the format "3" is *IEEE float*. To import the file into LibreTime you would need to covert it to a basic PCM wav file using a tool like Audacity.

The "Invalid WAVE file." error doesn't really help catch this kind of situation. I'll update the code to log the actual error to aid debugging this type of issue.

It would be even nicer if the wave module would actually support such files. There is a [python bug](https://bugs.python.org/issue1144504) about this but it seems to have been rejected. If this affects a large number of file we should probably look into implementing some kind of workaround.
